src/flatc.cpp

```diff
@@ -550,6 +549,15 @@ FlatCOptions FlatCompiler::ParseFromCommandLineArguments(int argc,
         options.binary_files_from = options.filenames.size();
       } else if (arg == "--proto") {
         opts.proto_mode = true;
+        if (!IsValidCodeGenerator(arg)) {
```

Comment:
shouldn't this all be removed if the new codegenerator is added in flatc_main.cpp?

Reply:
So we would need opts.proto_mode to be set to true. Given the way flat parsing is set up, if `--proto` is specified, only the `else if` block with `--proto` is evaluated. However, we need to add the generator to `options.generators` to actually use the proto generator.

So that's why I put the same logic when we parse `--proto` and line 670.

Reply:
Now to think about it, it might be better to move the `--proto` parsing logic into the `else` block at line 670 so they share the same code generator set up logic.

Reply:
Done